### ISSUE TITLE
Indent all keyvals

### ISSUE THREAD

**Issue by creativecreature** [charmbracelet/log#118]
Hi, thanks a lot for creating this package!

I have been playing around with it a bit today, but I have not been able to find a way that ensures that the `keyvals` always gets indented.

To illustrate what I mean, I've created this program:

```go
package main

import (
	"os"

	"github.com/charmbracelet/log"
)

const (
	userId  = "31c572eb-abb8-4907-b4ad-9c66799c4895"
	orderId = "6b3508f9-cfbe-41d7-8c4d-d4c399bee604"
)

func main() {
	logger := log.New(os.Stdout)
	logger.Info("Successfully placed order.",
		"user_id", userId,
		"order_id", orderId,
	)
}

```

Running itprints the message as a single line like this:

<img width="1230" alt="Screenshot 2024-04-16 at 20 45 25" src="https://github.com/charmbracelet/log/assets/12787673/3e30d027-ac7b-4a5f-9a33-514a79f0c0d5">

However, I looked a bit at the code, and saw that there was an indentation separator. Is there a way to ensure that it gets applied to all keyvals without having to add newlines to each value like this:

```go
logger.Info("Successfully placed order.",
	"user_id", userId+"\n",
	"order_id", orderId+"\n",
)
```

<img width="430" alt="Screenshot 2024-04-16 at 20 49 01" src="https://github.com/charmbracelet/log/assets/12787673/8f294ebc-bff1-43f0-a336-b8c0c2efa7ea">

**Comment by maaslalani** [comment]
Perhaps we can add an option that allows you to do something like?

```
log.New().Multiline(true)
```

On the other hand @creativecreature, is there a particular reason you want to avoid adding new lines if that gives you the desired result?

**Comment by creativecreature** [comment]
I think that your proposed option would be great! I'm currently using a similar approach, though it does become a little tedious over time. The new lines does not really give me the result  I'm after, but it's better than having a really long line! With a `Multiline` option I think both the aesthetics and readability would be improved if the key and value would be aligned on the same line

**Comment by decentral1se** [comment]
Yes, chiming in here to say that this would be a huuuuuge usability win, especially for programs which have large traces with keyvals.

**Comment by tonglil** [comment]
Any leads on creating a formatter that will use Logfmt for the level and message, but then serialize keyvals into formatted json and print them each on their own line?

Something like:
```
INFO Successfully placed order.
{
  "user_id": "xxxxxxxxx",
  "order_id", "xxxxxxxxx",
}
INFO ...
```
